### PR TITLE
Rename target specific files for consistency

### PR DESCRIPTION
**Pull request by jkotas**
- .CoreCLR.cs: CoreCLR-runtime specific files
- .CoreRT.cs: CoreRT-runtime specifc files
- .Uap.cs: Universal app platform specific files. Both UAP (Universal App Platform) and UWP (Universal Windows Platform) have been used interchangeably. I have picked UAP for the file names since it is the name used in the RID (e.g. uap101). There are number of places where UWP is used in the comments and such - I have not done anything about those.

Fixes issue #9875.

### CODE REVIEW

**Comment by weshaggard** [comment]
Could you also please update the guidelines at https://github.com/dotnet/corefx/blob/master/Documentation/coding-guidelines/project-guidelines.md#code-file-naming-conventions?

**Comment by jkotas** [comment]
@weshaggard Updated the documentation

**Comment by jkotas** [comment]
@dotnet-bot test Innerloop CentOS7.1 Debug Build and Test please

**Comment by jkotas** [comment]
@dotnet-bot test Innerloop CentOS7.1 Release Build and Test please

**Comment by jkotas** [comment]
@weshaggard I made some more documentation updates. PTLA

File: src/Common/src/System/IO/DriveInfoInternal.Uap.cs [hunk 1]
```diff
@@ -12,7 +12,7 @@ internal static partial class DriveInfoInternal
     {
         internal static string[] GetLogicalDrives()
         {
-            // Getting volume information isn't supported on WinRT
+            // Getting volume information isn't supported on UWP
```
Comment (weshaggard) [inline src/Common/src/System/IO/DriveInfoInternal.Uap.cs]: Any particular reason you used UAP for file name and UWP for the comment?

Comment (jkotas) [inline src/Common/src/System/IO/DriveInfoInternal.Uap.cs]: I have been going back and forth about the right name ... fixed.

File: Documentation/coding-guidelines/project-guidelines.md [hunk 2]
```diff
@@ -38,7 +38,7 @@ For each unique configuration needed for a given library project a configuration
 - `$(Configuration) -> $(OSGroup)_$(TargetGroup)_$(ConfigurationGroup)`
 - `$(OSGroup) -> [Empty]/AnyOS* | Windows | Linux | OSX | FreeBSD`
 - `$(TargetGroup) -> [Empty]* | <PackageTargetFramework> | <PackageTargetRuntime> | <Version> | <PackageTargetFramework><PackageTargetRuntime>`
- - `$(PackageTargetFramework) -> net46x | netstandard1.x | netcoreapp1.x | uap10.x | netcore50` 
+ - `$(PackageTargetFramework) -> net46x | netstandard1.x | netcoreapp1.x | uap10x | netcore50`
```
Comment (weshaggard) [inline Documentation/coding-guidelines/project-guidelines.md]: this should be uap10.x

File: src/System.IO.FileSystem/src/System.IO.FileSystem.csproj [hunk 3]
```diff
@@ -251,25 +251,25 @@
       <Link>Common\Interop\Windows\Interop.SetErrorMode.cs</Link>
     </Compile>
   </ItemGroup>
-  <!-- Windows : Win32 + WinRT -->
+  <!-- Windows : UWP - Win32 + WinRT -->
```
Comment (weshaggard) [inline src/System.IO.FileSystem/src/System.IO.FileSystem.csproj]: There are a couple more comments with UWP.

File: Documentation/coding-guidelines/project-guidelines.md (hunk 2 above)
Comment (jkotas) [inline Documentation/coding-guidelines/project-guidelines.md]: Fixed.

**Pull request merged**